drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodListDialogFragment.kt

```diff
@@ -149,7 +146,7 @@ class PaymentMethodListDialogFragment :
             }
             PaymentMethodTypes.SUPPORTED_PAYMENT_METHODS.contains(paymentMethod.type) -> {
                 Logger.d(TAG, "onPaymentMethodSelected: payment method is supported")
-                protocol.showComponentDialog(paymentMethodsListViewModel.getPaymentMethod(paymentMethod))
+                protocol.showComponentDialog(paymentMethodsListViewModel.getPaymentMethod(paymentMethod.index))
```

Comment:
Why do we need to pass the index here instead of the whole `paymentMethod`?

Reply:
we don't need the whole payment method we just need the index to get the corresponding paymentMethod

Reply:
Yes but shouldn't the view model decide which part of the `paymentMethod` it needs, instead of the fragment?

Reply:
tbh I hate the way this method is done 😅, but we can remove this thing and replace it with something like that 
```
protocol.showComponentDialog(dropInViewModel.getPaymentMethods()[paymentMethod.index])
```
and remove `getPaymentMethod()` from the view model

Reply:
By using `dropInViewModel.getPaymentMethods()[paymentMethod.index]` you are assuming that:
- `dropInViewModel.paymentMethods` is the same as `paymentMethodsListViewModel.paymentMethods`
- The `index` inside `PaymentMethodModel` matches the index of the corresponding payment method inside the list in the view model

While both these points are true now, any change we do to the logic in either view models might break this flow. Because currently the view model is the one that maps `PaymentMethod`s to `PaymentMethodModel`s, so if the fragment calls `dropInViewModel.getPaymentMethods()[paymentMethod.index]` it's basically mapping the `PaymentMethodModel` back to a `PaymentMethod` itself which can easily cause bugs.

WDYT?